### PR TITLE
[Enhancement][Storage] refactor Comparison Predicates

### LINKED ISSUE #10379: [Enhancement][Storage] refactor Comparison Predicates
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

remove macro and refine code

### Solution

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
# Proposed changes

Issue Number: close #10379

------ dataset: ssb-flat, concurrency: 1 ------
sql\time(ms)\parallel_num       1
          before  after
q1      206.6    210.8
q2      185.2    187.4
q3      422.0    252.4
q4      284.0    293.0
q5      203.2    202.8
q6      125.4    126.0
q7      1367.6    1345.8
q8      255.0    218.8
q9      178.8    146.4
q10     131.4    122.8
q11     551.6    528.0
q12     234.6    237.6
q13     187.0    183.2

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/comparison_predicate.h [hunk 1]
```diff
@@ -17,46 +17,475 @@
...
+        } else if constexpr (PT == PredicateType::LE) {
+            return lhs <= rhs;
+        } else if constexpr (PT == PredicateType::GT) {
+            return lhs > rhs;
+        } else if constexpr (PT == PredicateType::GE) {
+            return lhs >= rhs;
+        }
+    }
+
+    constexpr bool _is_range() const {
```
Comment (yiguolei) [inline be/src/olap/comparison_predicate.h:295]: Not very accurate.     PredicateType could be
    EQ = 1,
    NE = 2,
    LT = 3,
    LE = 4,
    GT = 5,
    GE = 6,
    IN_LIST = 7,
    NOT_IN_LIST = 8,
    IS_NULL = 9,
    IS_NOT_NULL = 10,
    BF = 11, // BloomFilter

better use > < >= <=

File: be/src/olap/comparison_predicate.h [hunk 2]
```diff
@@ -17,46 +17,475 @@
...
+                                     .get_data();
+
+            _base_evaluate<true, is_and>(&nested_column, &null_map, sel, size, flags);
+        } else {
+            _base_evaluate<false, is_and>(&column, nullptr, sel, size, flags);
+        }
+    }
+
+    template <bool is_nullable, typename TArray, typename TValue>
+    void _base_loop(uint16_t size, bool* flags,
```
Comment (mrhhsg) [inline be/src/olap/comparison_predicate.h:367]: `_base_loop` -> `_base_loop_vec`

File: be/src/olap/comparison_predicate.h [hunk 3]
```diff
@@ -17,46 +17,475 @@
...
+            if constexpr (is_nullable) {
+                flags[i] = !(*null_map)[i] && _operator(data_array[i], value);
+            } else {
+                flags[i] = _operator(data_array[i], value);
+            }
+        }
+    }
+
+    template <bool is_nullable, bool is_and, typename TArray, typename TValue>
+    void _base_loop(const uint16_t* sel, uint16_t size, bool* flags,
```
Comment (mrhhsg) [inline be/src/olap/comparison_predicate.h:380]: `_base_loop` -> `_base_loop_and_or` maybe better.

**Review by yiguolei — DISMISSED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**